streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -702,15 +720,34 @@ private ProcessorTopology build(String applicationId, Set<String> nodeGroup) {
      * Get the names of topics that are to be consumed by the source nodes created by this builder.
      * @return the unmodifiable set of topic names used by source nodes, which changes as new sources are added; never null
      */
-    public synchronized Set<String> sourceTopics(String applicationId) {
-        Set<String> topics = new HashSet<>();
-        for (String topic : sourceTopicNames) {
+    public synchronized Set<String> sourceTopics() {
+        Set<String> topics = maybeDecorateInternalSourceTopics(sourceTopicNames);
+        return Collections.unmodifiableSet(topics);
+    }
+
+    private Set<String> maybeDecorateInternalSourceTopics(final Set<String> sourceTopics) {
+        return maybeDecorateInternalSourceTopics(sourceTopics.toArray(new String[sourceTopics.size()]));
+    }
+
+    private Set<String> maybeDecorateInternalSourceTopics(String ... sourceTopics) {
+        final Set<String> decoratedTopics = new HashSet<>();
+        for (String topic : sourceTopics) {
             if (internalTopicNames.contains(topic)) {
-                topics.add(applicationId + "-" + topic);
+                decoratedTopics.add(decorateTopic(topic));
             } else {
-                topics.add(topic);
+                decoratedTopics.add(topic);
             }
         }
-        return Collections.unmodifiableSet(topics);
+        return decoratedTopics;
+    }
+
+    private String decorateTopic(String topic) {
```

Comment:
Thanks for pointing out. I am trying to make trunk closer to 0.10.0 in https://github.com/apache/kafka/pull/1736. As for this change in particular, I will make it consistent with #1736 by removing the parameter but directly depend on the `applicationId` reference.